File: libsessionpass/OtsvCode.cpp
//
// Created by EnKdev on 27.11.2024.
//

#include "OtsvCode.h"
#include "HttpHelper.h"

#include <map>


std::string EnKdev::SessionPass::OtsvCode::requestOtsvCode(int codeFlag) {
    std::map<std::string, std::string> pairs = {
        {"cf", std::to_string(codeFlag)}
    };

    std::string res = HttpHelper::httpPost("https://enkdev.xyz/cdn/php/sessionpass/generateOtsvCode.php", pairs);
    return res;
}

bool EnKdev::SessionPass::OtsvCode::compareOtsvCode(const std::string &otsvCode, int codeFlag) {
    bool isSame = false;

    if (!otsvCode.empty()) {
        std::map<std::string, std::string> pairs = {
            {"code", otsvCode},
            {"cf", std::to_string(codeFlag)}
        };

        std::string res = HttpHelper::httpPost("https://enkdev.xyz/cdn/php/sessionpass/compareCode.php", pairs);

        if (res == "true") {
            isSame = true;
        }
    } else {
        std::cout << "otsvCode is null, can't check." << std::endl;
    }

    return isSame;
}

File: EnKdev.SessionPass.Delphi/OtsvCode.pas
unit OtsvCode;

interface

uses
  SysUtils, Classes,
  System.Net.HttpClient, System.Net.URLClient,
  System.Net.Mime;

// typedecl
// Declare TOtsvCode as a class and define its functions.
type
  TOtsvCode = class
  public
    class function RequestOtsvCode(CodeFlag: Integer): string; static;
    class function CompareOtsvCode(const otsvCode: string; CodeFlag: Integer): Boolean; static;
  end;

implementation

uses
  HttpHelper;

{ TOtsvCode }

class function TOtsvCode.RequestOtsvCode(CodeFlag: Integer): string;
var
  Params: TStringList;
  Response: TBytes;
  Utf8Encoding: TUTF8Encoding;
begin
  Params := TStringList.Create;

  try
    Params.Add('cf=' + CodeFlag.ToString);
    Response := THttpHelper.HttpPost('https://enkdev.xyz/cdn/php/sessionpass/generateOtsvCode.php', Params);

    Utf8Encoding := TUTF8Encoding.Create;

    try
      Result := Utf8Encoding.GetString(Response);
    finally
      Utf8Encoding.Free;
    end;
  finally
    Params.Free;
  end;
end;

class function TOtsvCode.CompareOtsvCode(const otsvCode: string; CodeFlag: Integer): Boolean;
var
  Params: TStringList;
  Response: TBytes;
  Utf8Encoding: TUTF8Encoding;
  ResponseStr: string;
begin
  Params := TStringList.Create;
  Result := false;

  if (OtsvCode <> '') and (CodeFlag >= 0) then
  begin
    try
      Params.Add('code=' + OtsvCode);
      Params.Add('cf=' + CodeFlag.ToString);

      Response := THttpHelper.HttpPost('https://enkdev.xyz/cdn/php/sessionpass/compareCode.php', Params);

      Utf8Encoding := TUTF8Encoding.Create;

      try
        ResponseStr := Utf8Encoding.GetString(Response);

        if ResponseStr = 'true' then
          Result := True
        else
          Result := False;
      finally
        Utf8Encoding.Free;
      end;
    finally
      Params.Free;
    end;
  end
  else
    Writeln('otsvCode is null, can''t check.');

end;

end.
